projects/RabbitMQ.Client/client/api/ConnectionFactory.cs

```diff
@@ -188,6 +189,15 @@ public sealed class ConnectionFactory : ConnectionFactoryBase, IAsyncConnectionF
 
         // just here to hold the value that was set through the setter
         private Uri _uri;
+        private readonly ArrayPool<byte> _memoryPool;
+
+        /// <summary>
+        /// The memory pool used for allocating buffers. Default is <see cref="MemoryPool{T}.Shared"/>.
+        /// </summary>
+        public ArrayPool<byte> MemoryPool
```

Comment:
This seems to use a different pattern than what regular settings do. Most settings in this class have get and set and the default value is assigned as the default of the property why deviate from this?

Reply:
In my original PR, the property had a setter. But then it was changed by @lukebakken

Reply:
I can revert, no problem.

Reply:
I don't care too much I just wanted to point out the difference of how optional and default settings are applied